src/main/java/org/isf/telemetry/daemon/TelemetryDaemon.java

```diff
@@ -0,0 +1,155 @@
+/*
+ * Open Hospital (www.open-hospital.org)
+ * Copyright © 2006-2022 Informatici Senza Frontiere (info@informaticisenzafrontiere.org)
+ *
+ * Open Hospital is a free and open source software for healthcare data management.
+ *
+ * This program is free software: you can redistribute it and/or modify
+ * it under the terms of the GNU General Public License as published by
+ * the Free Software Foundation, either version 3 of the License, or
+ * (at your option) any later version.
+ *
+ * https://www.gnu.org/licenses/gpl-3.0-standalone.html
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License
+ * along with this program. If not, see <http://www.gnu.org/licenses/>.
+ */
+package org.isf.telemetry.daemon;
+
+import java.time.LocalDateTime;
+
+import org.isf.generaldata.ConfigurationProperties;
+import org.isf.generaldata.GeneralData;
+import org.isf.menu.manager.Context;
+import org.isf.telemetry.manager.TelemetryManager;
+import org.isf.telemetry.model.Telemetry;
+import org.isf.telemetry.util.TelemetryUtils;
+import org.isf.utils.ExceptionUtils;
+import org.isf.utils.exception.OHException;
+import org.isf.utils.time.TimeTools;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+public class TelemetryDaemon extends ConfigurationProperties implements Runnable {
+
+	private static final Logger LOGGER = LoggerFactory.getLogger(TelemetryDaemon.class);
+	private static final String FILE_PROPERTIES = "telemetry.properties";
+	private static final int DEFAULT_DELAY = 30; // seconds
+
+	// Singleton instance
+	private static volatile TelemetryDaemon instance;
+	private Thread telemetryThread;
```

Comment:
Given this is a `private` class variable and is `started` in `start()`, shouldn't there be a `stop` somewhere in this class?

Reply:
you mean that it SHOULD be there a method to stop `telemetryThread`?

Reply:
I would assume so and it should be invoked at shutdown.   If there isn't then the class variable can be local to the method.

Reply:
I do recall that threads should never be stopped, the thread is always running until it accomplishes its tasks, then it stops itself and the `running` variable is used here only to perform or not perform some operations during the thread life

Reply:
Added stop() method and one convenient method, please check

Reply:
OK